gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1468,9 +1468,34 @@ def test_{{ method_name }}_rest_no_http_options():
 {{ async_method_prefix }}def test_{{ method_name }}_{{ test_name }}_{{transport_name}}():
     client = {{ get_client(service=service, is_async=is_async) }}(
         credentials={{ get_credentials(is_async=is_async) }},
+        transport="{{ transport_name }}",
     )
 
-    # Mock the actual call within the gRPC stub, and fake the request.
+    # Mock the actual call, and fake the request.
+    {% if 'rest' in transport %}
+    with mock.patch.object(
+            type(client.transport.{{ method.transport_safe_name|snake_case }}),
+            '__call__') as call, mock.patch.object(type(client.transport._session), 'request') as req:
+        req.return_value = mock.Mock()
+        req.return_value.status_code = 200
+        {% if method.void %}
+        return_value = None
+        json_return_value = ''
+        {% elif method.lro %}
+        return_value = operations_pb2.Operation(name='operations/spam')
+        json_return_value = json_format.MessageToJson(return_value)
+        {% else %}
+        return_value = {{ method.output.ident }}()
+        {% if method.output.ident.is_proto_plus_type %}
+        # Convert return value to protobuf type
+        return_value = {{ method.output.ident }}.pb(return_value)
+        {% endif %}
+        json_return_value = json_format.MessageToJson(return_value)
+        {% endif %}
+
+        req.return_value.read = mock.AsyncMock(return_value=json_return_value)
```

Comment:
this doesn't seem right for the case of sync and should be guarded with a conditional statement.

We may instead want something like:
```
{% if is_async %}
req.return_value.read = mock.AsyncMock(return_value=json_return_value)
{% else %}
response_value.content = json_return_value.encode
{% endif %}

```

Reply:
If the test case isn't failing because of this then there may be something going wrong here.

Reply:
Fixed in b6f76934ae12d207c26dce86f94b499f8eb6d258